File: Spring 2020/TripleSum/Java/main.java
import java.util.Hashtable;
import java.util.Arrays;

public class TripleSum_Program {
    public static void main(String []args) {
       System.out.println("Hello World");
    }

   public static boolean two_sum(int []input, int target){
      Hashtable<Integer, Integer> lookUpTable = new Hashtable<Integer, Integer>(); 
      for (Integer i : input) {
          if (lookUpTable.containsKey(i)) {
              return true;
          }
      }

      return false;
   }

   public static boolean triple_sum(int []input, int target) {
       for (int i = 0; i < input.length; i++) {
           if (two_sum(Arrays.copyOfRange(input, 0, i) + Arrays.copyOfRange(input, i + 1, input.length)), target - input[i]) {
               return true;
           }
       }
   }

   public static boolean triple_sum_brute_force(int []input, int target) {
       for (int i = 0; i < input.length; i++) {
           for (int j = i + 1; j < input.length; j++) {
               for (int k = j + 1; k < input.length; k++) {
                   if (target == input[i] + input[j] + input[k]) {
                       return true;
                   }
               }
           }
       }
       return false;
   }
 }

File: Spring 2020/TripleSum/Python/main.py
def triple_sum_brute_force(a, b):
    for i in range(len(a)):
        for j in range(i + 1, len(a)):
            for k in range(j + 1, len(a)):
                if (a[i] + a[j] + a[k] == b):
                    return True

    return False

def triple_sum(a,b):
    for index, element in enumerate(a):
        if two_sum(a[:index] + a[index + 1:], b - element):
            return True
    return False

def two_sum(a, b):
    look_up_table = {}
    for i in a:
        if i in look_up_table:
            return look_up_table[i]
        look_up_table[b - i] = i
    return False
